apps/admin-gui/src/app/shared/components/dialogs/update-application-form-dialog/update-application-form-dialog.component.ts

```diff
@@ -17,13 +18,13 @@ export interface UpdateApplicationFormDialogData {
 export class UpdateApplicationFormDialogComponent implements OnInit {
   entity: string;
   applicationForm: ApplicationForm;
-  moduleName: string;
   initialState: string;
   extensionState: string;
   embeddedState: string;
   loading = false;
   theme: string;
   autoRegistrationEnabled: boolean;
+  formArray: FormArray<FormControl>;
```

Comment:
When using  form controls always define the type itself. This way we get help and type safety from typescript.
Make sure to include type when creating form controls in this file. Also remove any casting to string (`control.value as string`) as it is no longer needed.

Otherwise good job with the `FormArray`!
```suggestion
  formArray: FormArray<FormControl<string>>;
```